substrate/frame/contracts/uapi/src/api.rs

```diff
@@ -0,0 +1,759 @@
+
+	/// Stores the *free* balance of the current account into the supplied buffer.
+	///
+	/// If the available space in `output` is less than the size of the value a trap is triggered.
+	///
+	/// # Parameters
+	///
+	/// - `output`: A reference to the output data buffer to write the balance.
+	fn balance(output: &mut &mut [u8]);
+
+	/// Stores the current block number of the current contract into the supplied buffer.
+	///
+	/// If the available space in `output` is less than the size of the value a trap is triggered.
+	///
+	/// # Parameters
+	///
+	/// - `output`: A reference to the output data buffer to write the block number.
+	fn block_number(output: &mut &mut [u8]);
+
+	/// Make a call to another contract.
+	///
+	/// This is equivalent to calling the newer version of this function with
+	/// `flags` set to [`CallFlags::ALLOW_REENTRY`]. See the newer version for documentation.
+	#[deprecated(note = "Deprecated, use newer version instead")]
+	fn call(
+		callee: &[u8],
+		gas: u64,
+		value: &[u8],
+		input_data: &[u8],
+		output: Option<&mut [u8]>,
+	) -> Result;
+
+	/// Make a call to another contract.
+	///
+	/// Equivalent to the newer [`Self::call_v2`] version but works with
+	/// *ref_time* Weight only
+	fn call_v1(
+		flags: CallFlags,
+		callee: &[u8],
+		gas: u64,
+		value: &[u8],
+		input_data: &[u8],
+		output: Option<&mut [u8]>,
+	) -> Result;
+
+	/// Call (possibly transferring some amount of funds) into the specified account.
+	///
+	/// # Parameters
+	///
+	/// - `flags`: See [`CallFlags`] for a documentation of the supported flags.
+	/// - `callee`: The address of the callee. Should be decodable as an `T::AccountId`. Traps
+	///   otherwise.
+	/// - `ref_time_limit`: how much *ref_time* Weight to devote to the execution.
+	/// - `proof_size_limit`: how much *proof_size* Weight to devote to the execution.
+	/// - `deposit`: The storage deposit limit for instantiation. Should be decodable as a
+	///   `Option<T::Balance>`. Traps otherwise. Passing `None` means setting no specific limit for
+	///   the call, which implies storage usage up to the limit of the parent call.
+	/// - `value`: The value to transfer into the contract. Should be decodable as a `T::Balance`.
+	///   Traps otherwise.
+	/// - `input`: The input data buffer used to call the contract.
+	/// - `output`: A reference to the output data buffer to write the call output buffer. If `None`
+	///   is provided then the output buffer is not copied.
+	///
+	/// # Errors
+	///
+	/// An error means that the call wasn't successful output buffer is returned unless
+	/// stated otherwise.
+	///
+	/// - [CalleeReverted][`crate::ReturnErrorCode::CalleeReverted]: Output buffer is returned.
+	/// - [CalleeTrapped][`crate::ReturnErrorCode::CalleeTrapped]
+	/// - [TransferFailed][`crate::ReturnErrorCode::TransferFailed]
+	/// - [NotCallable][`crate::ReturnErrorCode::NotCallable]
+	#[deprecated(note = "Unstable, use `call_v1` instead")]
```

Comment:
Should be replaced with the general warning above. I think it is kind of obvious that we can fall back to an older version.